osu.Game/Database/RealmAccess.cs

```diff
@@ -902,7 +902,14 @@ private void createBackup(string backupFilename)
                 {
                     using (var source = storage.GetStream(Filename, mode: FileMode.Open))
                     using (var destination = storage.GetStream(backupFilename, FileAccess.Write, FileMode.CreateNew))
+                    {
+                        // source may not exist.
+                        if (source == null)
+                            return;
```

Comment:
Any reason to still be creating the backup if the source doesn't exist?

Reply:
It doesn't create it due to the early `return`. Would you want a precondition check for file existing?

Reply:
Am I missing something? I just tested in an isolated app:
```csharp
using (var fs = new FileStream("/home/smgi/Desktop/wangs.txt", FileMode.CreateNew, FileAccess.Write))
    return;
```
And it creates the file.

Reply:
Ah I get you now.